MC/config/PWGDQ/external/generator/GeneratorCocktail.C

```diff
@@ -30,9 +30,19 @@ class GeneratorCocktail_class : public Generator
   bool importParticles() override
   {
     for (auto& g : *mEntries) {
+      int nPart = mParticles.size();
       g->importParticles();
-      for (auto& p : g->getParticles())
+      for (auto p : g->getParticles()) {
```

Comment:
You could do
```c++
for (auto& p : g->getParticles()) {
  mParticles.push_back(p);
  auto& pEdit = mParticles.back();
  // do what's needed with pEdit
}
That should be one copy less

Reply:
ok, I implemented it in the way you suggested.